File: app/src/main/java/com/prykhodev/android_dictionary/model/util/AppBindings.java
package com.prykhodev.android_dictionary.model.util;

import com.prykhodev.android_dictionary.model.Meaning;
import com.prykhodev.android_dictionary.ui.main.searchFragment.MeaningListAdapter;

import java.util.List;

import androidx.databinding.BindingAdapter;
import androidx.recyclerview.widget.RecyclerView;

public class AppBindings {
    @BindingAdapter("app:items")
    public static void setMeanings(RecyclerView view, List<Meaning> meanings) {
        if(meanings != null) {
            MeaningListAdapter adapter = (MeaningListAdapter) view.getAdapter();
            if(adapter != null) adapter.submitList(meanings);
        }
    }

    private AppBindings() {}
}

File: app/src/main/java/com/prykhodev/android_dictionary/model/util/AppBindings.kt
package com.prykhodev.android_dictionary.model.util

import androidx.databinding.BindingAdapter
import androidx.recyclerview.widget.RecyclerView
import com.prykhodev.android_dictionary.model.Meaning
import com.prykhodev.android_dictionary.ui.main.searchFragment.MeaningListAdapter

@BindingAdapter("app:items")
fun setMeanings(view: RecyclerView, meanings: List<Meaning>?) {
    meanings?.let { (view.adapter as MeaningListAdapter).submitList(it) }
}